src/main/java/com/google/flourbot/datastorage/FirebaseDataStorage.java

```diff
@@ -45,8 +45,8 @@ public FirebaseDataStorage() {
     // Create a query to find a macro named macroName belonging to userEmail
     Query query =
         db.collection(COLLECTION_NAME)
-            .whereEqualTo(USER_IDENTIFIER, userEmail)
-            .whereEqualTo(MACRO_IDENTIFIER, macroName);
+            .whereEqualTo(MACRO_IDENTIFIER, macroName)
+            .whereArrayContains(SCOPE_IDENTIFIER, userEmail);
```

Comment:
Once again, you may want to create a composite index on Firebase.

Reply:
Sure, if we aren't going with using the RoomID, I will add the index in.